### DISCUSSION TITLE
Keybinding for "Switch Focus To Editor"

### DISCUSSION THREAD

**Discussion by DanEEStar** [zed-industries/zed#6826] [Feature Requests]
I can map a keybinding to go to the terminal like `"terminal_panel::ToggleFocus"`. 
But I could not map a keybinding to go back to the editor from the terminal.
There are different `ActivatePane...` actions. 

A workaround is the binding to `["workspace::ActivatePane", 0]`.
But I want to go back to the editor to the last pane I was in before...

**Comment by mrnugget** [comment]
I have this in my `keymap.json`:

```json
  {
    "context": "Terminal",
    "bindings": {
      "ctrl-w t": "terminal_panel::ToggleFocus"
    }
  }
```

Does that work?

**Reply by DanEEStar** [reply]
This works!

**Reply by DanEEStar** [reply]
@mrnugget I am just these days working through your interpreter book by the way. I really like it.
Nice coincidence. 😃

**Reply by mrnugget** [reply]
Glad to hear it! :)

**Comment by FabianHofmann** [comment]
Thanks for the discussion, I was able to achieve the expected behavior for the terminal pane, however I would like my shortcut to work globally for any pane. There I am struggling to make it work for the assistant pane. I use ctrl-e for switching back to the editor pane, so I would expect the following keymap entry to work.

```json
  {
    "context": "AssistantPanel",
    "bindings": {
      "ctrl-shift-e": "assistant::ToggleFocus"
    }
  },
```

This does not do anything, unfortunately. Any help would be appreciated :)

**Comment by dv01sw** [comment]
There seems to be a default keycode now, it works for toggling between editor and terminal:

     CTRL + `

**Reply by sonalkarmakar** [reply]
<key>Ctrl</key>+<key>`</key> closes the terminal when switching focus to the editor. Upon checking the Keymap, I found that <key>Ctrl</key>+<key>`</key> is set to `terminal panel: toggle` in `Workspace` context, and <key>Ctrl</key>+<key>~</key> is set to `workspace: new terminal` in `Workspace` context.

For my keyboard (and I assume for most others), <key>`</key> and <key>~</key> are on the same physical key, so triggering one also triggers the other. Is there a shortcut or setting that switches without closing the terminal?

Update: I solved with the following keybinds. I had to remove the existing ones for the keystrokes.
```json
  {
    "context": "Workspace",
    "unbind": {
      "ctrl-`": "terminal_panel::ToggleFocus"
    }
  },
  {
    "context": "Terminal",
    "bindings": {
      "ctrl-`": "editor::ToggleFocus"
    }
  }
```

**Comment by lulungsatrioprayuda** [comment]
i used zed , usually i used ctrl  +shift + E to focus on file navbar, press again to focus on ediot after that i use ctrl + j to focus or even open the terminal in zed
